reddit/script.js

```diff
@@ -0,0 +1,67 @@
+document.addEventListener('DOMContentLoaded', function () {
+    const postsContainer = document.getElementById('posts-container');
+    const subredditInput = document.getElementById('subreddit-input');
+    const fetchButton = document.getElementById('fetch-button');
+
+    function fetchRedditPosts(subreddit) {
+        fetch(`https://www.reddit.com/r/${subreddit}/.json`)
+            .then((response) => response.json())
+            .then((data) => {
+                postsContainer.innerHTML = '';
+
+                data.data.children.slice(0, 10).forEach((post) => {
+                    const postElement = document.createElement('div');
+                    postElement.classList.add('post');
+
+                    const postTitle = document.createElement('div');
+                    postTitle.classList.add('post-title');
+                    postTitle.textContent = post.data.title;
+
+                    if (post.data.media && post.data.media.reddit_video) {
+                        const videoElement = document.createElement('video');
+                        videoElement.src = post.data.media.reddit_video.fallback_url;
+                        videoElement.controls = true;
+                        videoElement.width = `300`;
+                        videoElement.height = `200`;
+                        postElement.appendChild(videoElement);
+                    } else if (post.data.url && post.data.url.endsWith('.jpg')) {
```

Comment:
I don't think this block of code from lines 27-33 is necessary. When I comment this out, everything seems to work fine. You handle images using thumbnails in the next block too.